### DISCUSSION TITLE
Change the way create-astro downloads templates

### DISCUSSION THREAD

**Discussion by Mikkel-T** [withastro/roadmap#82] [Proposal]
After having spent a lot of time looking into ways of mitigating all the `degit` issues that are occurring in `create-astro`, I have started to wonder if it is even a good idea to support using other examples than our own. This thought came to me after having checked out some other `create-*` packages where I saw that almost none of them supported any other templates than their own ones.

If you take a look at [Vite's documentation](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) about this, you can see that they only support their own examples, and then have [instructions on how to do it with community templates](https://vitejs.dev/guide/#community-templates). [Next.js does almost the same](https://nextjs.org/docs/api-reference/create-next-app#why-use-create-next-app), except that they don't tell you how you can use other templates as well.

I am suggesting that we should do the same thing with `create-astro`, and limit it only to the official examples/templates while still giving instructions about how to download other templates. This would first and foremost make us able to remove `degit` and create our own downloading logic. Of course there are also a couple of ways to do that, we could either go the Vite way and make use of `fs` to just straight up copy the files, or we could go the Next.js way and download a tar of the code and extract it. There are many options, but I definitely think that it would be worth it in the long run instead of having to deal with all the issues that comes with `degit`.

I am willing to make a PR that implements this, but it would need to be discussed a bit before that.

**Comment by matthewp** [comment]
We do have some external examples that we've encouraged people to use with `create-astro` in the past, like this one: https://github.com/cassidoo/shopify-react-astro.

**Reply by Mikkel-T** [reply]
I am aware of that, and of course it wouldn't be ideal if people used `create-astro` with examples like that and it stopped supporting them, but I still do believe that a mix of what Vite and Next are doing is the way to go. With the way `create-astro` works now, you can copy any repo from GitHub, GitLab, Bitbucket and Sourcehut and in my opinion, that's should be what tools like `degit` are used for, not framework-specific project scaffolding tools like `create-astro`. If you again look at a project like Next.js, they have about 300 examples in their repo which can all be downloaded via the `create-next-app` command, while Vite as mentioned above has instructions for how to use `degit` to download community templates.
Again, the main reason I made this discussion is all the issues we have with `degit`, and that this was the best solution I could find after some searching. But like with this example, I know I might have missed some things that I don't normally use and therefor didn't think would be a problem which is why it is really nice to get some other eyes on the idea!
